### PR TITLE
OPENEUROPA-1501: Implement configuration UI for Authentication.

### PR DESCRIPTION
**Pull request by nagyad**
## OPENEUROPA-1501
Provide the Authentication module with a UI so that users with appropriate permissions can edit and change module configuration.

### CODE REVIEW

File: src/Form/AuthenticationSettingsForm.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form['protocol'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Application authentication protocol'),
+      '#default_value' => $this->config(static::CONFIGNAME)->get('protocol'),
```
Comment (voidtek) [inline src/Form/AuthenticationSettingsForm.php]: "#description" could be useful.
This remark is relevant to other fields too.

File: tests/features/configure_authentication.feature [hunk 2]
```diff
@@ -0,0 +1,30 @@
+@api
+Feature: Authentication
+  As the site manager
+  I need to be able to configure the settings
+
+  Background:
+    Given I am logged in as a user with the "administer authentication configuration" permission
+
+  @DrupalLogin @BackupCasConfigs
```
Comment (voidtek) [inline tests/features/configure_authentication.feature]: I think that is a duplicate and can create confusion in the configuration.

File: src/Form/AuthenticationSettingsForm.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    ];
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $this->config(static::CONFIGNAME)
+      ->set('protocol', $form_state->getValues()['protocol'])
```
Comment (upchuk) [inline src/Form/AuthenticationSettingsForm.php]: @nagyad FormStateInterface has a `getValue()` method that allows to ask for a single submitted value. Either use that or use `getValues()` into an array variable at the top and access its keys here. Preferably the first one.

File: tests/Behat/AuthenticationContext.php [hunk 4]
```diff
@@ -4,26 +4,53 @@
...
    */
   public function setConfigDrupalLogin(): void {
-    $this->setConfig('cas.settings', 'forced_login.enabled', FALSE);
+    $this->configContext->setConfig('cas.settings', 'forced_login.enabled', FALSE);
+  }
+
+  /**
+   * Configures the CAS module to use Cas login.
+   *
+   * For some reason this is being cached.
```
Comment (upchuk) [inline tests/Behat/AuthenticationContext.php]: Dude...

Comment (drupol) [reply tests/Behat/AuthenticationContext.php]: ?

File: tests/features/configure_authentication.feature [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+Feature: Authentication
+  As the site manager
+  I need to be able to configure the settings
+
+  Background:
+    Given I am logged in as a user with the "administer authentication configuration" permission
+
+  @DrupalLogin @BackupAuthConfigs
+  Scenario: Configure Authentication settings
+    Given I am on "admin/config/system/oe_authentication"
```
Comment (imanoleguskiza) [inline tests/features/configure_authentication.feature]: Same scenario as https://github.com/openeuropa/oe_webtools/pull/24#discussion_r246788190

**Pull request merged**